cinema/views.py

```diff
@@ -1 +1,67 @@
-# write views here
+from rest_framework import viewsets
+from cinema.models import Genre, Actor, CinemaHall, Movie, MovieSession
+from cinema.serializers import (
+    GenreSerializer,
+    ActorSerializer,
+    CinemaHallSerializer,
+    MovieSerializer,
+    MovieSessionSerializer,
+    MovieSessionListSerializer,
+    MovieRetrieveSerializer,
+    MovieListSerializer,
+    MovieSessionRetrieveSerializer,
+)
+
+
+class GenreViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+
+class CinemaHallViewSet(viewsets.ModelViewSet):
+    queryset = CinemaHall.objects.all()
+    serializer_class = CinemaHallSerializer
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
+
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.action in ("list", "retrieve"):
+            queryset = Movie.objects.prefetch_related("actors", "genres")
```

Comment:
I think you will make two requests to SQL server here since you use Movie manager in line 32 and line 37 ( I may be wrong since querysets are lazy mfs) .  Instead you could use queryset you created in line 35 since it has queryser there already

Reply:
First query set will be used for method POST PUT etc, where it is not required to have field actors and genre. Another queryset definitely will make 3 request, but it is fine, it fixed amount